pymc3/gp/gp.py

```diff
@@ -13,49 +13,51 @@
 
 __all__ = ['GP', 'sample_gp']
 
+
 class GP(Continuous):
     """Gausian process
-    
+
     Parameters
     ----------
     mean_func : Mean
         Mean function of Gaussian process
     cov_func : Covariance
         Covariance function of Gaussian process
     X : array
-        Grid of points to evaluate Gaussian process over. Only required if the 
+        Grid of points to evaluate Gaussian process over. Only required if the
         GP is not an observed variable.
     sigma : scalar or array
         Observation standard deviation (defaults to zero)
     """
     def __init__(self, mean_func=None, cov_func=None, X=None, sigma=0, *args, **kwargs):
-        
-        if mean_func is None:
-            self.M = Zero()
-        else:
-            if not isinstance(mean_func, Mean):
-                raise ValueError('mean_func must be a subclass of Mean')
-            self.M = mean_func
-            
+
         if cov_func is None:
             raise ValueError('A covariance function must be specified for GPP')
         if not isinstance(cov_func, Covariance):
             raise ValueError('cov_func must be a subclass of Covariance')
         self.K = cov_func
-        
+
+        if mean_func is None:
+            self.M = Zero(input_dim=cov_func.active_dims)
+        else:
+            if not isinstance(mean_func, Mean):
+                raise ValueError('mean_func must be a subclass of Mean')
+            self.M = mean_func
+
+
```

Comment:
It was so the default mean function could take parameters that the user would have set when they instantiated the covariance function